### PR TITLE
Github action for collating all assets into a single CSV file

### PR DESCRIPTION
**Pull request by buhrmann**
Does some preprocessing so that values in a single column are type-consistent (no mixing of strings, floats or objects). Also homogenises missing data representation.

Can easily be extended in the future to write different outputs (json, parquet...), and to add other kinds of preprocessing (proper numeric representation of model sizes e.g.).

The collated CSV file will automatically be commited to `resources/all_assets.csv`.

Feel free to squash the merge, as I needed some intermediate commits to fix initial attempts at the github action.

### CODE REVIEW

**Review by rishibommasani — COMMENTED** [review]
This looks clean to me, and no major comments! Will let @dilarasoylu or @percyliang check as well if they have time, since I am not too familiar with Github actions and want to make sure we haven't missed anything.

File: scripts/collate.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    """Store all assets together in a single tabular CSV file."""
+    paths = ASSET_PATH.glob("*.yaml")
+    assets = [asset for fp in paths for asset in yaml.safe_load(open(fp))]
+    df = DataFrame.from_records(assets)
+
+    # Make data consistent within columns (cannot have scalar and dicts in same column)
+    df = df.apply(lambda ser: ser.apply(scalar))
+
+    # Cleanup missing data representations
+    df = df.replace("none", None).replace("unknown", None)
```
[the remark this answers is not in the document]
Comment (percyliang) [reply scripts/collate.py:35]: Could we keep these as is instead of replacing with None?

[the remark this answers is not in the document]
Comment (buhrmann) [reply scripts/collate.py:35]: Good point. Unfortunately I think most if not all tabular formats (DataFrames like pandas or polars, databases, file formats like Apache Arrow or parquet etc.) only allow a single unique missing value "sentinel". If you're lucky you get a distinction between nan ("invalid" numerical value) and None (missing). But you can't have a string like "unknown" in a numerical column. I could be more precise perhaps and distinguish between numerical and categorical/text data and keep "unknown" where possible. I'll give that a try 👍🏼

File: scripts/collate.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+ASSET_PATH = Path("./assets")
+
+
+def scalar(val):
+    """Clean scalar values whenever possible."""
+    if not isinstance(val, dict):
+        return val
+
+    if "explanation" in val:
```
[the remark this answers is not in the document]
Comment (buhrmann) [reply scripts/collate.py:16]: Again, this is so as not to mix nested data with floats or text in the same column. If necessary, or in a future iteration, one could also keep the explanation in a separate, parallel column to that holding the actual values.

File: scripts/collate.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (buhrmann) [reply scripts/collate.py]: Changed in below commit. I didn't realize all variables other than `created_date` where essentially categorical/text data. There are no numerical variables at the moment for example. In a future version I'll try to preprocess things like `size`, `monthly_active_users`, `training_emissions` etc. into proper numerical values (without units or comments in the fields), so they can be more easily used to filter or sort the dataset

**Review by percyliang — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**